### PR TITLE
feat: integrate the latest updates

### PR DESCRIPTION
**Pull request by huaweigu**
## Summary
Align with the latest developments from the private repository [circlefin:entry-point-v0.6](https://github.com/circlefin/buidl-wallet-contracts/tree/entry-point-v0.6).

### Changeset
* resolve all compiler warnings and enable `deny_warnings` moving forward
* address all reported findings from the second audit conducted by CS

### Checklist
- [x] Did you add new tests and confirm all tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `docs` folder)
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint`) and fix any issues?
- [x] Did you run formatter (`yarn format:check`) and fix any issues (`yarn format:write`)?

## Testing
* tests have been added to validate the audit fixes

## Documentation
n/a

### CODE REVIEW

File: script/006_DeployAddressBookPlugin.s.sol [hunk 1]
```diff
@@ -1,44 +0,0 @@
-/*
```
Comment (huaweigu) [inline script/006_DeployAddressBookPlugin.s.sol:1]: Does not apply to this branch.

File: script/008_DeployWeightedWebauthnMultisigPlugin.s.sol [hunk 2]
```diff
@@ -1,54 +0,0 @@
-/*
- * Copyright 2024 Circle Internet Group, Inc. All rights reserved.
```
Comment (huaweigu) [inline script/008_DeployWeightedWebauthnMultisigPlugin.s.sol:2]: Does not apply to this branch.

File: script/011_DeployTokenCallbackPlugin.s.sol [hunk 3]
```diff
@@ -1,44 +0,0 @@
-/*
```
Comment (huaweigu) [inline script/011_DeployTokenCallbackPlugin.s.sol:1]: Does not apply to this branch.

File: src/msca/6900/v0.7/plugins/v1_0_0/addressbook/AddressBookPlugin.sol [hunk 4]
```diff
@@ -1,444 +0,0 @@
-/*
```
Comment (huaweigu) [inline src/msca/6900/v0.7/plugins/v1_0_0/addressbook/AddressBookPlugin.sol:1]: Does not apply to this branch.

File: src/msca/6900/v0.7/plugins/v1_0_0/multisig/BaseMultisigPlugin.sol [hunk 5]
```diff
@@ -1,189 +0,0 @@
-/*
```
Comment (huaweigu) [inline src/msca/6900/v0.7/plugins/v1_0_0/multisig/BaseMultisigPlugin.sol:1]: Does not apply to this branch.

File: src/msca/6900/v0.7/plugins/v1_0_0/multisig/BaseWeightedMultisigPlugin.sol [hunk 6]
```diff
@@ -1,512 +0,0 @@
-/*
- * Copyright 2024 Circle Internet Group, Inc. All rights reserved.
```
Comment (huaweigu) [inline src/msca/6900/v0.7/plugins/v1_0_0/multisig/BaseWeightedMultisigPlugin.sol:2]: Does not apply to this branch.

File: src/msca/6900/v0.7/plugins/v1_0_0/multisig/IWeightedMultisigPlugin.sol [hunk 7]
```diff
@@ -1,190 +0,0 @@
-/*
```
Comment (huaweigu) [inline src/msca/6900/v0.7/plugins/v1_0_0/multisig/IWeightedMultisigPlugin.sol:1]: Does not apply to this branch.

File: src/msca/6900/v0.7/plugins/v1_0_0/multisig/README.md [hunk 8]
```diff
@@ -1,45 +0,0 @@
-## Weighted Multisig Plugin
```
Comment (huaweigu) [inline src/msca/6900/v0.7/plugins/v1_0_0/multisig/README.md:1]: Does not apply to this branch.

File: src/msca/6900/v0.7/plugins/v1_0_0/multisig/WeightedWebauthnMultisigPlugin.sol [hunk 9]
```diff
@@ -1,529 +0,0 @@
-/*
```
Comment (huaweigu) [inline src/msca/6900/v0.7/plugins/v1_0_0/multisig/WeightedWebauthnMultisigPlugin.sol:1]: Does not apply to this branch.

File: src/msca/6900/v0.7/plugins/v1_0_0/utility/DefaultTokenCallbackPlugin.sol [hunk 10]
```diff
@@ -1,121 +0,0 @@
-/*
```
Comment (huaweigu) [inline src/msca/6900/v0.7/plugins/v1_0_0/utility/DefaultTokenCallbackPlugin.sol:1]: Does not apply to this branch.

**Review by dlim-circle — APPROVED** [review]

**Pull request merged**